webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala

```diff
@@ -174,53 +230,62 @@ class ZarrStreamingController @Inject()(
-                             cxyz: String): Action[AnyContent] =
+                             coordinates: String): Action[AnyContent] =
     Action.async { implicit request =>
-      for {
-        annotationSource <- remoteWebknossosClient.getAnnotationSource(accessToken, urlOrHeaderToken(token, request)) ~> NOT_FOUND
-        relevantToken = if (annotationSource.accessViaPrivateLink) Some(accessToken)
-        else urlOrHeaderToken(token, request)
-        layer = annotationSource.getAnnotationLayer(dataLayerName)
-
-        // ensures access to volume layers if fallback layer with equal name exists
-        result <- layer match {
-          case Some(annotationLayer) =>
-            remoteTracingstoreClient
-              .getRawZarrCube(annotationLayer.tracingId, mag, cxyz, annotationSource.tracingStoreUrl, relevantToken)
-              .map(Ok(_))
-          case None =>
-            rawZarrCube(annotationSource.organizationName, annotationSource.datasetName, dataLayerName, mag, cxyz)
-        }
-      } yield result
+      ifIsAnnotationLayerOrElse(
+        token,
+        accessToken,
+        dataLayerName,
+        ifIsAnnotationLayer = (annotationLayer, annotationSource, relevantToken) =>
+          remoteTracingstoreClient
+            .getRawZarrCube(annotationLayer.tracingId,
+                            mag,
+                            coordinates,
+                            annotationSource.tracingStoreUrl,
+                            relevantToken)
+            .map(Ok(_)),
+        orElse = (annotationSource) =>
+          rawZarrCube(annotationSource.organizationName, annotationSource.datasetName, dataLayerName, mag, coordinates)
+      )
     }
 
   private def rawZarrCube(
       organizationName: String,
       datasetName: String,
       dataLayerName: String,
       mag: String,
-      cxyz: String,
+      coordinates: String,
   )(implicit m: MessagesProvider): Fox[Result] =
     for {
       (dataSource, dataLayer) <- dataSourceRepository.getDataSourceAndDataLayer(organizationName,
                                                                                 datasetName,
                                                                                 dataLayerName) ~> NOT_FOUND
-      (c, x, y, z) <- ZarrCoordinatesParser.parseDotCoordinates(cxyz) ?~> "zarr.invalidChunkCoordinates" ~> NOT_FOUND
+      //  (c, x, y, z)
+      parsedCoordinates <- ZarrCoordinatesParser.parseNDimensionalDotCoordinates(coordinates) ?~> "zarr.invalidChunkCoordinates" ~> NOT_FOUND // TODO: change error message
       magParsed <- Vec3Int.fromMagLiteral(mag, allowScalar = true) ?~> Messages("dataLayer.invalidMag", mag) ~> NOT_FOUND
       _ <- bool2Fox(dataLayer.containsResolution(magParsed)) ?~> Messages("dataLayer.wrongMag", dataLayerName, mag) ~> NOT_FOUND
-      _ <- bool2Fox(c == 0) ~> "zarr.invalidFirstChunkCoord" ~> NOT_FOUND
+      _ <- bool2Fox(parsedCoordinates.head == 0) ~> "zarr.invalidFirstChunkCoord" ~> NOT_FOUND
```

Comment:
head can throw an exception. access with headOption instead
```suggestion
      channelCoordinate <- parsedcoordinates.headOption ?~> NOT_FOUND
      _ <- bool2Fox(channelCoordinate == 0) ~> "zarr.invalidFirstChunkCoord" ~> NOT_FOUND
```